src/lib.rs

```diff
@@ -55,11 +55,33 @@ pub enum Either<L, R> {
     Right(R),
 }
 
-macro_rules! either {
+/// Evaluate the provided expression for both [`Either::Left`] and [`Either::Right`].
+///
+/// This macro is useful in cases where both sides of [`Either`] can be interacted with
+/// in the same way even though the don't share the same type.
```

Comment:
We could toss in a pseudo signature like this 
```suggestion
/// in the same way even though the don't share the same type.
///
/// Syntax: `either::for_both!(` *expression* `,` *pattern* `=>` *expression* `)`
```

The same way I did it here: https://docs.rs/matches/0.1.9/matches/macro.matches.html

I don't know if we have a developing other consistent way to do this for macros, any other standard would be good too

Reply:
Do you want me to add something like this before merging?